syntax = "proto3";

option go_package = "github.com/vectorch-ai/scalellm;scalellm";
package llm.proto;

message ModelCard {
  // The model identifier, which can be referenced in the API endpoints.
  optional string id = 1;

  // The Unix timestamp (in seconds) when the model was created.
  optional uint32 created = 2;

  // the object type, which is always "model".
  optional string object = 3;  

  // the organization that owns the model.
  optional string owned_by = 4 [json_name = "owned_by"];
}

message ListRequest {
  // The model identifier.
  // string model = 1;
}

message ListResponse {
  // The list of models.
  repeated ModelCard data = 1;
}

service Models {
  // Lists the currently available models, and provides basic information
  // about each one such as the owner and availability.
  rpc List(ListRequest) returns (ListResponse) {}
}